File: src/main/java/com/nao4j/subtrol/service/QuantityService.java
package com.nao4j.subtrol.service;

import com.nao4j.subtrol.document.internal.Quantity.QuantityType;
import org.springframework.security.access.annotation.Secured;

import java.util.Set;

import static com.nao4j.subtrol.security.Roles.ADMIN;
import static com.nao4j.subtrol.security.Roles.USER;

/**
 * Сервис для работы с размерностями.
 */
public interface QuantityService {

    /**
     * Получить множество поддерживаемых размерностей.
     */
    @Secured({USER, ADMIN})
    Set<QuantityType> getAll();

}

File: src/main/kotlin/com/nao4j/subtrol/service/QuantityService.kt
package com.nao4j.subtrol.service

import com.nao4j.subtrol.model.Quantity.QuantityType
import com.nao4j.subtrol.security.ADMIN
import com.nao4j.subtrol.security.USER
import org.springframework.security.access.annotation.Secured

/**
 * Сервис для работы с размерностями.
 */
interface QuantityService {

    /**
     * Получить множество поддерживаемых размерностей.
     */
    @Secured(value = [USER, ADMIN])
    fun getAll(): Set<QuantityType>

}
